.github/workflows/workflow.yml

```diff
@@ -94,14 +83,28 @@ jobs:
       # Ensure Dune can build itself
       - run: opam exec -- ./dune.exe build -p dune --profile dune-bootstrap
 
-      - name: run test suite
-        run: opam exec -- ocaml .github/workflows/ci.ml test
-        if: env.SKIP_TEST != 'true'
+      - name: Install deps on Unix
+        run: |
+          opam install . --deps-only --with-test
+          opam exec -- make dev-deps
+        if: ${{ matrix.os != 'windows-latest' && matrix.skip_test == false }}
+
+      - name: Install deps on Win32
+        run: opam install ./dune-configurator.opam --deps-only --with-test
+        if: ${{ matrix.os == 'windows-latest' && matrix.skip_test == false }}
+
+      - name: Run test suite on Unix
+        run: opam exec -- make test
+        if: ${{ matrix.os != 'windows-latest' && matrix.skip_test == false }}
+
+      - name: Run test suite on Win32
+        run: opam exec -- make test-windows
+        if: ${{ matrix.os == 'windows-latest' && matrix.skip_test == false }}
 
-      - name: test source is well formatted
+      - name: Test source is well formatted
         run: opam exec -- make fmt
-        if: env.OCAML_VERSION == '4.12.0' && env.OS == 'ubuntu-latest'
+        if: ${{ matrix.os == 'ubuntu-latest' && matrix.ocaml-compiler == '4.13.x' }}
```

Comment:
The version changed, is there a reason for this? (but both should return the same results)